src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/ProjectView.xaml.cs

```diff
@@ -165,11 +165,22 @@ protected override void OnKeyUp(KeyEventArgs e)
                             {
                                 // Search for the best version
                                 NuGetVersion rangeBestVersion = userRequestedVersionRange.FindBestMatch(versions);
-                                bool isBestOption = rangeBestVersion.ToString() == _versions.Items[_versions.SelectedIndex].ToString();
+                                DisplayVersion selectedVersion = (DisplayVersion)_versions.Items.CurrentItem;
+                                bool isBestOption = rangeBestVersion.ToString() == selectedVersion.Version.ToString();
                                 if (isBestOption)
                                 {
-                                    PackageDetailControlModel.SelectedVersion = new DisplayVersion(userRequestedVersionRange, rangeBestVersion, additionalInfo: null, isVulnerable: false);
-                                    _versions.Text = comboboxText;
+                                    // Add vulnerable/deprecated labels to non floating/range versions
+                                    if (userRequestedVersionRange.OriginalString.StartsWith("(", StringComparison.OrdinalIgnoreCase) ||
+                                        userRequestedVersionRange.OriginalString.StartsWith("[", StringComparison.OrdinalIgnoreCase) ||
```

Comment:
Just thinking out loud here, I'm undecided whether I recommend a change or not. Generally, I don't like string manipulation, I prefer strong data type checking. `VersionRange` has `HasLowerBound` and `IsMinInclusive` booleans, plus another set for upper/max. Those could be checked, rather than checking `OriginalString`.
Alternatively, maybe `.OriginalString[0] == '('` avoids needing to pass in the string comparison, but I'm not sure if the analyzer will flag that as using the current culture, and suggesting to use a culture aware API instead.